pipeline.py

```diff
@@ -69,31 +93,121 @@ class TextProcessingPipeline:
+            multiple_tagged_tokens = []
+
+            for processed_token in processed_tokens:
+                cleaned_tokens.append(processed_token.get_cleaned())
+                single_tagged_tokens.append(processed_token.get_single_tagged())
+                multiple_tagged_tokens.append(processed_token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # Gets rid of '-\n's that break up words, so we won't get 'кот-' 'орый' as example (Should be: 'который')
+        text = raw_text.replace('-\n', '')
+
+        result = Mystem().analyze(text)
+
+        morphological_tokens = []
+
+        morph = pymorphy2.MorphAnalyzer()
+
+        for token_info in result:
+            original_word = token_info['text']
+
+            if not re.match(r'[а-яА-Яa-zA-Z]', original_word):
+                continue
+
+            morphological_token = MorphologicalToken(original_word=original_word)
+
+            if 'analysis' not in token_info:
+                continue
+
+            if not token_info['analysis']:
+                continue
+
+            if not {'lex', 'gr'}.issubset(set(token_info['analysis'][0])):
+                continue
+
+            morphological_token.normalized_form = token_info['analysis'][0]['lex']
+            morphological_token.tags_mystem = token_info['analysis'][0]['gr']
+
+            parsed_word = morph.parse(original_word)[0]
+            morphological_token.tags_pymorphy = parsed_word.tag
+
+            morphological_tokens.append(morphological_token)
+
+        return morphological_tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+
+    pathlib_path_to_validate = Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not pathlib_path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    file_ids = []
+
+    for file_path in pathlib_path_to_validate.iterdir():
+        file_name = file_path.name
+
+        match = re.match(r'\d+', file_name)
+
+        if not match:
+            raise InconsistentDatasetError
+
+        file_id = int(match.group(0))
+
+        if file_id not in file_ids:
+            file_ids.append(file_id)
+
+    file_ids = sorted(file_ids)
+
+    last_file_id = 0
+
+    for file_id in file_ids:
+        if not last_file_id and file_id != 1 or file_id - last_file_id > 1 or \
+                not (pathlib_path_to_validate / f'{file_id}_raw.txt').is_file() or \
+                not (pathlib_path_to_validate / f'{file_id}_meta.json').is_file():
+            raise InconsistentDatasetError
```

Comment:
please enrich each raise statement with a message describing the problem. this message should be passed as an argument during error instantiation.

For example, I want to see a message "There are not meta or raw files for an article ID: 3"